storage/innobase/log/log0log.cc

```diff
@@ -1092,12 +1103,14 @@ lsn_t log_t::write_buf() noexcept
 
     ut_ad(base + (write_lsn_offset & (WRITE_TO_BUF - 1)) == lsn);
     write_to_log++;
+    if (arch_sys)
+      arch_sys->log_sys()->wait_archiver(lsn);
 
     if (resizing != RETAIN_LATCH)
       latch.wr_unlock();
```

Comment:
`Arch_Sys::archiver_wait()` may release and reacquire `log_sys.latch`. This would break the assumption that is being made in `log_write_and_flush()`.

I think that it would be cleaner to introduce data members to `log_t` related to archiving the log, such as `clone_lsn` and `clone_log` (similar to `resize_lsn` and `resize_log`), and introduce a `log_t::clone_write_buf()` as well. In that way, cloning the log would simply rely on the existing group commit mechanism and there would be fewer context switches and no potential race condition scenarios.